feelpp/feel/feelmesh/filters.hpp

```diff
@@ -53,6 +78,44 @@ make_elements_wrapper()
     using elt_wrapper_t = typename MeshTraits<MeshType>::elements_reference_wrapper_type;
     return std::make_shared<elt_wrapper_t>();
 }
+template <typename RangeT>
+using range_elementtype_t = tuple_element_t<0, RangeT>;
+template <typename RangeT>
+constexpr size_type range_elementtype_v = range_elementtype_t<RangeT>::value;
+template <typename RangeT>//, std::enable_if_t < std::is_same_v < std::tuple_element<1, RangeT>,std::tuple_element<2, RangeT>>,int> = 0 >
+using range_iterators_t = tuple_element_t<1, RangeT>;
+template <typename RangeT>
+using range_container_t = tuple_element_t<3, RangeT>;
+
+template <typename RangeT>
+inline constexpr bool is_range_v = is_tuple_v<RangeT>;//std::is_base_of_v<std::input_iterator_tag, typename std::iterator_traits<range_iterators_t<RangeT>>::iterator_category>;
+
+
+//!
+//! @return the worldcomm associated to the mesh stored in the range
+//! @ingroup Mesh
+//!
+template <typename RangeT, std::enable_if_t<is_range_v<RangeT>, int> = 0>
+WorldComm const& worldComm( RangeT const& range )
+{
+    return boost::unwrap_ref( *begin(range) ).mesh()->worldComm();
```

Comment:
will crash if the range is empty on one process. Maybe return the shared pointer if non null else a null pointer

Reply:
probably the new struct that you develop for the range will fix that

Reply:
oui